### PR TITLE
Fix unit tests warnings that prolong test time

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix unit tests warnings that prolong test time
- add/fix props as required by component
- removed other tests which are not relevant

#### Ticket Link
none

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: tests/components/channel_notifications_modal/describe.test.jsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ describe('components/channel_notifications_modal/NotificationSection', () => {
     });
 
     test('should match snapshot, on NEVER', () => {
-        const props = {...baseProps, memberNotifyLevel: NotificationLevels.NEVER};
+        const props = {...baseProps, memberNotifyLevel: NotificationLevels.NONE};
```
Comment (saturninoabril) [inline tests/components/channel_notifications_modal/describe.test.jsx]: It's NONE not NEVER under `NotificationLevels` constant.

File: tests/components/channel_notifications_modal/expand_view.test.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {shallow} from 'enzyme';
 
 import {NotificationLevels, NotificationSections} from 'utils/constants.jsx';
 
-import ExpandView from 'components/channel_notifications_modal/components/collapse_view.jsx';
+import ExpandView from 'components/channel_notifications_modal/components/expand_view.jsx';
```
Comment (saturninoabril) [inline tests/components/channel_notifications_modal/expand_view.test.jsx]: Fix incorrect path.

File: tests/components/edit_channel_purpose_modal.test.jsx [hunk 3]
```diff
@@ -146,7 +146,7 @@ describe('comoponents/EditChannelPurposeModal', () => {
                 ctrlSend={false}
                 requestStatus={RequestStatus.STARTED}
                 onModalDismissed={jest.fn()}
-                actions={{patchCHannel: jest.fn()}}
+                actions={{patchChannel: jest.fn()}}
```
Comment (saturninoabril) [inline tests/components/edit_channel_purpose_modal.test.jsx]: Fix spelling.

File: tests/components/edit_post_modal.test.jsx [hunk 4]
```diff
@@ -67,11 +67,6 @@ describe('components/EditPostModal', () => {
         expect(wrapper).toMatchSnapshot();
     });
 
-    it('should match without editingPost', () => {
-        const wrapper = shallow(createEditPost({editingPost: {}}));
-        expect(wrapper).toMatchSnapshot();
-    });
```
Comment (saturninoabril) [inline tests/components/edit_post_modal.test.jsx]: Remove this test as `props. editingPost` is required.

File: tests/components/post_view/date_separator.test.jsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ import DateSeparator from 'components/post_view/date_separator.jsx';
 
 describe('components/post_view/DateSeparator', () => {
     test('should match snapshot', () => {
-        const wrapper = mountWithIntl(<DateSeparator date={'Fri Jan 12 2018 20:15:13 GMT+0800 (+08)'}/>);
+        const wrapper = mountWithIntl(<DateSeparator date={new Date('Fri Jan 12 2018 20:15:13 GMT+0800 (+08)')}/>);
```
Comment (saturninoabril) [inline tests/components/post_view/date_separator.test.jsx]: props.date accepts Date object

File: tests/components/sidebar/sidebar.test.jsx [hunk 6]
```diff
@@ -157,27 +157,6 @@ describe('component/sidebar/sidebar_channel/SidebarChannel', () => {
...
-        wrapper = shallow(
-            <Sidebar
-                {...{
-                    ...defaultProps,
-                    currentUser: null,
-                }}
-            />
-        );
-        expect(wrapper).toMatchSnapshot();
-    });
```
Comment (saturninoabril) [inline tests/components/sidebar/sidebar.test.jsx]: Remove this test as currentTeam and currentUser is required.

File: tests/components/channel_notifications_modal/describe.test.jsx (hunk 1 above)
Comment (jespino) [reply tests/components/channel_notifications_modal/describe.test.jsx]: Still `NEVER` in the test description.

Comment (saturninoabril) [reply tests/components/channel_notifications_modal/describe.test.jsx]: Good catch! I'll update that.

**Pull request merged**